### ISSUE TITLE
Do not diagnose no-self-use or unused-argument for methods of generic structural types (Protocol[T])

### ISSUE THREAD

**Issue by jmehnle** [PyCQA/pylint#3885]
This is a follow-up to https://github.com/PyCQA/pylint/issues/3002, which addressed the basic issue, but there's a special case that is still broken: when a protocol class is a generic, i.e., it inherits not simply from `Protocol` but from `Protocol[T]` or some other generic form of `Protocol`.

### Steps to reproduce

```python
import typing_extensions as tex
from typing import TypeVar

T = TypeVar("T")

@tex.runtime
class Hasher(tex.Protocol[T]):
    """A hashing algorithm, e.g. :func:`hashlib.sha256`."""
    def update(self, bs: bytes):
        ...
    def digest(self) -> bytes:
        ...
```

### Current behavior

```
************* Module pylinttest
pylinttest.py:9:21: W0613: Unused argument 'bs' (unused-argument)
pylinttest.py:9:4: R0201: Method could be a function (no-self-use)
pylinttest.py:11:4: R0201: Method could be a function (no-self-use)
pylinttest.py:2:0: C0411: standard import "from typing import TypeVar" should be placed before "import typing_extensions as tex" (wrong-import-order)
```

### Expected behavior

No diagnostics.

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.7.9 (default, Sep  6 2020, 13:20:25)
[Clang 11.0.3 (clang-1103.0.32.62)]
```

**Comment by hippo91** [comment]
@jmehnle thanks for the report. It makes sense and i can reproduce it.
